### DISCUSSION TITLE
Need help Carousel Progress

### DISCUSSION THREAD

**Discussion by sarussss** [davidjerleke/embla-carousel#597] [Help]
I want to make a progress bar like on Swiper slider.
https://codesandbox.io/s/swiper-pagination-progress-forked-h67qpr
Embla Carousel https://codesandbox.io/s/embla-carousel-progress-vanilla-forked-tn6ht5
I want it to display like this at first https://tppr.me/UlhWG instead of https://tppr.me/QH9lL
Please, help me.

**Comment by davidjerleke** [comment]
Hi @sarussss,

You can change the [0 in the code here](https://codesandbox.io/s/embla-carousel-progress-vanilla-forked-tn6ht5?file=/src/js/progress-bar.js:124-125) which is the minimum scroll progress value to anything you want. Additionally, you can modify the progress function however you want to suit your needs.

Best,
David

**Reply by sarussss** [reply]
Hi @davidjerleke, Thanks for the answer.
I don't want to enter any number, that number needs to be calculated based on the fact that the initial display represents what percentage of the entire slider. Can you show me the best way to get that number? I need to use scrollSnapList?

**Reply by davidjerleke** [reply]
Hi @sarussss,

You can get almost all internal information with:

```ts
emblaApi.internalEngine()
``` 

Please have a look and you should be able to find something useful.

Best,
David

**Reply by sarussss** [reply]
@davidjerleke  Thank you, I will try it.
